src/utils.ts

```diff
@@ -0,0 +1,46 @@
+import { warn, watch } from 'vue'
+import { VueDriftWidgetConfig } from './types'
+
+const inBrowser: boolean = typeof window !== 'undefined'
+
+export const loadScript = (
+  id: string,
+  config: VueDriftWidgetConfig,
+  snippetVersion = '0.3.1',
+): void => {
+  if (!inBrowser) {
+    throw new Error('Window object is missing.')
+  }
+
+  if (window.drift) {
+    warn('Drift widget script has been loaded already.')
+    return
+  }
+
+  watch(
+    config,
+    (config) => {
+      // Dynamically load dependencies
+      if (config.loadScript && !hasScript()) {
+        const script = document.createElement('script')
+        script.innerText = `!function(){var t=window.driftt=window.drift=window.driftt||[];if(!t.init){if(t.invoked)return void(window.console&&console.error&&console.error("Drift snippet included twice."));t.invoked=!0,t.methods=["identify","config","track","reset","debug","show","ping","page","hide","off","on"],t.factory=function(e){return function(){var n=Array.prototype.slice.call(arguments);return n.unshift(e),t.push(n),t}},t.methods.forEach(function(e){t[e]=t.factory(e)}),t.load=function(t){var e=3e5*Math.ceil(new Date/3e5),n=document.createElement("script");n.type="text/javascript",n.async=!0,n.crossorigin="anonymous",n.src="https://js.driftt.com/include/"+e+"/"+t+".js";var o=document.getElementsByTagName("script")[0];o.parentNode.insertBefore(n,o)}}}(),drift.SNIPPET_VERSION="${snippetVersion}";`
+        document.body.appendChild(script)
+      }
+
+      // Dynamically initalize widget
+      if (config.enabled && hasScript && !window.drift?.hasInitialized) {
+        window.drift?.load(id)
+      }
```

Comment:
still believe here could be a race between adding the script and having it fully loaded... so it's better to also trigger enabling it when `load` event triggers for the previous script tag? or am I wrong and this can't be?

Reply:
The first block adds a script to the header which loads another async script. The trick here is that once we added the drift block, that adds the drift dependencies, the html is actually stopped and the site is blocked until the operation resolves. Only then the JS will be executed further. Works pretty well and I don't think we need to change anything here.